build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/release/AddVersionTask.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.gradle.internal.release;
+
+import com.github.javaparser.StaticJavaParser;
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.NodeList;
+import com.github.javaparser.ast.body.ClassOrInterfaceDeclaration;
+import com.github.javaparser.ast.body.FieldDeclaration;
+import com.github.javaparser.ast.body.VariableDeclarator;
+import com.github.javaparser.ast.expr.NameExpr;
+import com.github.javaparser.printer.lexicalpreservation.LexicalPreservingPrinter;
+import com.google.common.annotations.VisibleForTesting;
+
+import org.elasticsearch.gradle.Version;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.api.tasks.options.Option;
+import org.gradle.initialization.layout.BuildLayout;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.NavigableMap;
+import java.util.Optional;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+public class AddVersionTask extends AbstractVersionTask {
+    private static final Logger LOGGER = Logging.getLogger(AddVersionTask.class);
+
+    private Version version;
+    private boolean setCurrent;
+
+    @Inject
+    public AddVersionTask(BuildLayout layout) {
+        super(layout);
+    }
+
+    @Option(option = "add-version", description = "Specifies the version to add")
```

Comment:
Seems like this could just be called `version`. Calling `:addVersion --add-version` feels a bit redundant :)

Reply:
I tried that, but it conflicts with the `--version` gradle option to print out the gradle version info!

Reply:
Could we use something different than, eg `--value`? "version" is already explicit in the task name.

Reply:
The release automation PR runs the add & remove tasks together in the same line, which requires the options for add and remove versions to have different names. We could use the same option, but that would mean doing an add & remove would be two separate gradle invocations

Reply:
Or we could merge the two tasks into one, which you then give `add-version` and `remove-version` options to

Reply:
That seems reasonable. eg an `updateVersions` task?

Reply:
Yeah, I'm good with that.

Reply:
I've changed it to a single `updateVersions` task